containers/src/Kokkos_DualView.hpp

```diff
@@ -1070,10 +1058,10 @@ class DualView : public ViewTraits<DataType, Properties...> {
         return;
       }
     } else {
-      if (modified_flags(1) >= modified_flags(0)) {
-        resize_on_device(arg_prop);
-      } else {
+      if (modified_flags(0)) {
         resize_on_host(arg_prop);
+      } else {
+        resize_on_device(arg_prop);
```

Comment:
This actually changes the behavior doesn't it? For example: 

```c++
v.modify_host();
v.modify_device();
v.resize(....);
```

That did resize on device before, but now would on host or?

Reply:
My preference is that this would error out. It does already if you set the flag `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK`.

Reply:
The assumption for this pull request is that calling `modify_host` and `modify_device` in succession without `sync` isn't legal since we can't know anymore which side is supposed to be more recent.
If that is the case, we might just use the check guarded by `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` unconditionally as @stanmoore1 suggests.

Reply:
Checking unconditionally makes sense to me

Reply:
Let me propose that separately. We might just switch the default to `ON` instead.

Reply:
See https://github.com/kokkos/kokkos/pull/8021.

Reply:
Ok lets get that in independently ...